plugins/entityreference/behavior/OgBehaviorHandler.class.php

```diff
@@ -122,11 +122,7 @@ public function OgMembershipCrud($entity_type, $entity, $field, $instance, $lang
     foreach ($items as $item) {
       $gid = $item['target_id'];
 
-      // Must provide correct state in the event that approval is required.
-      if (empty($item['state']) && $entity_type == 'user' && !og_user_access($group_type, $gid, 'subscribe without approval', $entity) && !og_user_access($group_type, $gid, 'administer group')) {
```

Comment:
Can you please test this scenario:

1. Create a private group (that is with `og_access` module enabled and configured)
1. Allow in the OG permissions, to subscribe to group
1. As a non-admin user visit directly the subscribe URL
1. Try to subscribe 

What is the OgMembership->state you get?